experimental/terraform/README.md

```diff
@@ -1,42 +1,187 @@
-  * create a `terraform.tfvars` file
-  * invoke terraform
-1. run the following commands to retrieve values needed for your SCM:
-  * ```
-    echo `terraform output -raw event_handler_endpoint`
-    echo `terraform output -raw event_handler_secret`
+> :information_source: This README is intended to replace `/setup/README.md`. 
+> When it's time to promote the Terraform installer to become the preferred
+> installation method (hopefully soon!): 
+> * move the current contents of `/setup` to `/setup/deprecated`
+> * move the contents of this directory to `/setup`
+> * remove this note
+
+# Installation guide
+
+This guide describes how to set up Four Keys with your GitHub or GitLab project. The main steps are:
+
+1.  [Runing the setup script](#running-the-setup-script)
+1.  Integrating with your GitHub or Git Lab repo by:
+    1.  [Collecting changes data](#collecting-changes-data)
+    1.  [Collecting deployment data](#collecting-deployment-data)
+    1.  [Collecting incident data](#collecting-incident-data)
+
+## Before you begin
+
+1.  Install [GCloud SDK](https://cloud.google.com/sdk/install).
+1.  You must be owner on a Google Cloud project that has billing enabled. You may either use this project to house the architecture for the Four Keys, or you will be given the option to create new projects. If you create new projects, the original Google Cloud project will NOT be altered during set up, but the billing information from this parent project will be applied to any projects created.
+1. Install [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli).
+
+
+## Running the setup script
+
+1.  Once you have your Google Cloud project, run the following setup script from the top-level directory of this repository:
+
+    ```bash
+    gcloud config set project <PARENT_PROJECT_ID>
+    cd setup
+    ./setup.sh 2>&1 | tee setup.log
+    ```
+1.  Answer the setup script's questions:
+
+    *   Would you like to create a new Google Cloud Project for the four key metrics? (y/N)
+        * If you choose no, you will be asked later to provide info about the project you wish to use.
+    *   Choose the event sources to configure...
+        *   Which version control system are you using?
+            * Choose the appropriate option for your VCS, or choose "other" to skip VCS integration
+        * Which CI/CD system are you using?
+            * Choose the appropriate option for your CICD system, or choose "other" to skip CICD integration
+        * _(see `/README.md#extending-to-other-event-sources` to integrate event sources not available during setup)_
+    *   Would you like to generate mock data? (y/N)
+        *   If you select yes, a script will run through and send mock GitLab or GitHub events to your event-handler.  This will populate your dashboard with mock data.  The mock data will include the work "mock" in the source. You can generate mock data without using the setup script. See [Generating mock data](../readme.md).
```

Comment:
To exclude the mockdata from the dashboard, update the SQL script to filter out any source with the word mock in it.   `WHERE source not like "%mock"`

Reply:
added!